src/sentry/monitoring/queues.py

```diff
@@ -88,6 +108,60 @@ def get_queue_by_name(name):
             return queue
 
 
+queue_monitoring_cluster = redis.redis_clusters.get(CLUSTER_NAME)
+
+
+def is_queue_healthy(queue_name: str) -> bool:
+    # check if queue is healthy by pinging Redis
+    return not queue_monitoring_cluster.sismember(KEY_NAME, queue_name)
+
+
+def is_healthy(queue_size):
+    return queue_size < UNHEALTHY_QUEUE_SIZE_THRESHOLD
+
+
+def update_queue_stats(redis_cluster, unhealthy) -> None:
+    if not unhealthy:
+        return
+
+    with redis_cluster.pipeline(transaction=True) as pipeline:
+        pipeline.delete(KEY_NAME)
+        pipeline.sadd(KEY_NAME, *unhealthy)
+        # expire in 1min if we haven't checked in
+        pipeline.expire(KEY_NAME, 60)
```

Comment:
The problem here is that you are using sets, and you can only expire the whole set at once…

So if a *different* queue is already in that set, you explicitly `delete` it first. Even if you did not do that, bumping the `expire` means that existing set members will also get that TTL, so they may live longer than expected.

Instead of using a single set, rather just define a unique key, per queue, like `f"{KEY_NAME}:{queue_name}"`, and check for that.

Reply:
The TTL was just intended to make sure that this key eventually vanishes if we (for whatever reason) stop making updates. You're right though that there's no real reason we can't just use a key per queue.